src/router_core/router_core.c

```diff
@@ -101,7 +188,11 @@ void qdr_core_free(qdr_core_t *core)
     // Stop and join the thread
     //
     core->running = false;
-    sys_cond_signal(core->action_cond);
+    sys_mutex_lock(core->action_lock);
+    if (core->sleeping) {
```

Comment:
@kgiusti I forgot to mention that this PR contains an important optimization under load to reduce the signaling work of the worker threads ie to save signaling core if it isn't sleeping. 
This is beneficial from many point of views although it won't show up on benchmarks: please let me know if I should move it into a separate PR/jira/commit :+1: